weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallChat.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * Code to determine if a call is a chat call, and if so,
+ * load the message data and render it.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+
+import {LoadingDots} from '../../../../../LoadingDots';
+import {isWeaveRef} from '../../filters/common';
+import {ChatView} from '../PromptPage/ChatView';
+import {useCallAsChat} from '../PromptPage/hooks';
+import {TraceCallSchema} from '../wfReactInterface/traceServerClientTypes';
+import {CallSchema} from '../wfReactInterface/wfDataModelHooksInterface';
+
+type CallChatProps = {call: TraceCallSchema};
+
+// Does this look like a message object?
+export const isMessage = (message: any): boolean => {
+  if (isWeaveRef(message)) {
+    // Benefit of the doubt
+    return true;
+  }
+  if (!_.isPlainObject(message)) {
+    return false;
+  }
+  // TODO: Check for role: string?
```

Comment:
yeah, i would check to make sure it exactly confirms to `export type Message = {
  role: string;
  content?: string | MessagePart[];
  tool_calls?: ToolCall[];
};`